src/gpuarray_array.c

```diff
@@ -1008,15 +1009,18 @@ int GpuArray_concatenate(GpuArray *r, const GpuArray **as, size_t n,
 
   res_off = r->offset;
   res_dims = r->dimensions;
+  res_flags = r->flags;
+  r->flags &= ~(GA_C_CONTIGUOUS|GA_F_CONTIGUOUS);
```

Comment:
is it useful to change that in collapse? At the end, if we end up with 1
dim, check if the stride is elemsize. If so, consider contiguous.

This would prevent to do this optimization at everyplace that call extcopy.

On Thu, May 19, 2016 at 10:52 AM, abergeron <notifications@github.com>
wrote:

> In src/gpuarray_array.c
> <https://github.com/Theano/libgpuarray/pull/178#discussion_r63892962>:
>
> > @@ -1008,15 +1009,18 @@ int GpuArray_concatenate(GpuArray *r, const GpuArray **as, size_t n,
> >
> >    res_off = r->offset;
> >    res_dims = r->dimensions;
> > +  res_flags = r->flags;
> > +  r->flags &= ~(GA_C_CONTIGUOUS|GA_F_CONTIGUOUS);
>
> It will use 1d if not explicitly tagged as contiguous.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/libgpuarray/pull/178/files/7be54760f35fb3c7b2698d66c532d95528cf762d..04c713b21efa469ae4fa1e356f9b977eb39ceafb#r63892962>
>